File: AtividadesRafael/Capitulo8/Atividade02.py
def quadrado(num):
    return num * num

def fatorial(num):
    resultado = 1
    for i in range(1, num + 1):
        resultado *= i
    return resultado

while True:
    print("escolha uma opção:")
    print("1. calcular quadrado")
    print("2. calcular fatorial")
    print("3. sair")
    escolha = int(input("digite sua escolha: "))

    if escolha == 1 or escolha == 2:
        num = int(input("digite um número: "))
    
    if escolha == 1:
        print(f"quadrado de {num}: {quadrado(num)}")
    elif escolha == 2:
        print(f"fatorial de {num}: {fatorial(num)}")
    elif escolha == 3:
        print("saindo...")
        break
    else:
        print("opção inválida! tente novamente.")

//comparacoes entre as implementações:

python com if/elif/else: o python, com suas estruturas if/elif/else, proporciona uma solução mais simples
e compacta, especialmente para menus com poucas opções. a sintaxe de python também é mais legível e rápida
de escrever.

File: AtividadesRafael/Capitulo8/Atividade02.c
#include <stdio.h>

int quadrado(int num) {
    return num * num;
}

int fatorial(int num) {
    int resultado = 1;
    for (int i = 1; i <= num; i++) {
        resultado *= i;
    }
    return resultado;
}

int main() {
    int escolha, num;
    
    do {
        printf("escolha uma opcao:\n");
        printf("1. calcular quadrado\n");
        printf("2. calcular fatorial\n");
        printf("3. sair\n");
        printf("digite sua escolha: ");
        scanf("%d", &escolha);

        if (escolha == 1 || escolha == 2) {
            printf("digite um numero: ");
            scanf("%d", &num);
        }

        switch (escolha) {
            case 1:
                printf("quadrado de %d: %d\n", num, quadrado(num));
                break;
            case 2:
                printf("fatorial de %d: %d\n", num, fatorial(num));
                break;
            case 3:
                printf("saindo\n");
                break;
            default:
                printf("opcao invalida! tente novamente.\n");
        }
    } while (escolha != 3);
    
    return 0;
}


//comparação entre as implementações:

c com switch/case: a solução em c é mais estruturada e pode ser mais eficiente,
mas exige o uso de um switch para cada opção. o switch é bastante claro quando se tem várias opções,
mas o código fica maior.
